File: recursion/recursion-examples/nth_term.cs
/*
Note: This file is designed to be copied out and compiled on your machine. 
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file. 

To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Compile the program
4. Run the program
*/

using System;

namespace isaac_code_samples
{
    class NthTermRecursive
    {
        void Main(string[] args){
            int n = 6;
            int increment = 3;
            int result = NthTerm(n, increment);
            Console.WriteLine(String.Format("Term {0} is {1}", n.ToString(), result.ToString()));
        }

        public static int NthTerm(int n, int increment){
            //returns the nth term of a numerical sequence
            if (n == 1){
                return 1;
            }else{
                return increment + NthTerm(n-1, increment);
            }
        }
    }
}

File: recursion/recursion-examples/python/nth_term.py
# Raspberry Pi Foundation
# Developed to be used alongside Isaac Computer Science,
# part of the National Centre for Computing Education
# Usage licensed under CC BY-SA 4

def nth_term(n, increment):
    """Recursively calculates and returns the nth term of a sequence of integers"""
    if n == 1:
        return 1
    else:
        return increment + nth_term(n-1, increment)


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == '__main__':
    increment = 6
    n = 10
    result = nth_term(n, increment)
    print(f"The term is {result}")
